### PR TITLE
Fix trie reuse after Commit in atomic sync

### PR DESCRIPTION
**Pull request by darioush**
## Why this should be merged
plugin/evm code should honor the invariant specified for committing tries here:
https://github.com/ava-labs/coreth/blob/0eef20a5aca777b5177961bc349fdb0da99c2d6a/trie/trie.go#L41-L48

## How this works
Re-opens the trie from the database after commit.

## How this was tested
CI

### CODE REVIEW

File: plugin/evm/atomic_syncer.go [hunk 1]
```diff
@@ -107,6 +107,12 @@ func (s *atomicSyncer) onLeafs(keys [][]byte, values [][]byte) error {
 					return err
 				}
 			}
+			// Trie should be re-opened after committing.
+			trie, err := s.atomicTrie.OpenTrie(root)
```
Comment (patrick-ogrady) [inline plugin/evm/atomic_syncer.go:111]: Do we only need to do this if `isCommit`?

Comment (darioush) [reply plugin/evm/atomic_syncer.go:111]: The trie is always committed, `isCommit` determines whether the `trieDB` was committed hence requiring a commit to the versiondb.

**Review by patrick-ogrady — DISMISSED** [review]

File: plugin/evm/atomic_syncer.go [hunk 2]
```diff
@@ -107,6 +107,12 @@ func (s *atomicSyncer) onLeafs(keys [][]byte, values [][]byte) error {
 					return err
 				}
 			}
+			// Trie should be re-opened after committing.
```
Comment (aaronbuchwald) [inline plugin/evm/atomic_syncer.go:110]:
```suggestion
			// Trie must be re-opened after committing (not safe for re-use after commit)
```

File: plugin/evm/atomic_backend.go [hunk 3]
```diff
@@ -181,6 +181,11 @@ func (a *atomicBackend) initialize(lastAcceptedHeight uint64) error {
 				return err
 			}
 		}
+		// Trie should be re-opened after committing.
```
Comment (aaronbuchwald) [inline plugin/evm/atomic_backend.go:184]:
```suggestion
		// Trie must be re-opened after committing (not safe for re-use after commit)
```

**Review by aaronbuchwald — DISMISSED** [review]
optional suggestion to make comments more explicit - LGTM

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**